### PR TITLE
joplin 0.10.65 (new formula)

### PR DESCRIPTION
**Pull request by laurent22**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is a new formula for [joplin](http://joplin.cozic.net/).

### CODE REVIEW

File: Formula/joplin.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system bin/"joplin", "config", "editor", "subl"
+    assert_match "editor = subl", shell_output("#{bin}/joplin config")
```
Comment (woodruffw) [inline Formula/joplin.rb]: Use a regexp literal here:

```ruby
assert_match /editor = subl/, shell_output("#{bin}/joplin config")
```

**Review by woodruffw — APPROVED** [review]

**Comment by laurent22** [comment]
@woodruffw, thank you for reviewing the pull request. As there are a few commits on it, do I need to squash them or is it somehow done automatically?

**Comment by woodruffw** [comment]
> thank you for reviewing the pull request. As there are a few commits on it, do I need to squash them or is it somehow done automatically?

You don't have to -- I'll do it when merging.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by laurent22** [comment]
Just wanted to check the status of this pull request. Is there anything more I need to do to get it merged?

File: Formula/joplin.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class Joplin < Formula
+  desc "Joplin, a free, open source note taking and to-do application"
+  homepage "http://joplin.cozic.net"
```
Comment (ilovezfs) [inline Formula/joplin.rb]: http://joplin.cozic.net/

File: Formula/joplin.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system bin/"joplin", "config", "editor", "subl"
+    assert_match /editor = subl/, shell_output("#{bin}/joplin config")
```
Comment (ilovezfs) [inline Formula/joplin.rb]: `"editor = subl"`

File: Formula/joplin.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class Joplin < Formula
+  desc "Joplin, a free, open source note taking and to-do application"
```
Comment (ilovezfs) [inline Formula/joplin.rb]: We know it's open-source otherwise it wouldn't be in Homebrew. Also the description should not start with the formula's name.

**Review by ilovezfs — APPROVED** [review]
some minor stuff but looks good

**Comment by laurent22** [comment]
Thanks, I've updated the formula. For the test, I [was asked](https://github.com/Homebrew/homebrew-core/pull/20153/commits/c138e42beee3978f89bdaf4b1a98afbe071f0380) before to change to regex. I don't mind either way but please confirm which one it should be.

File: Formula/joplin.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
 require "language/node"
 
 class Joplin < Formula
-  desc "Joplin, a free, open source note taking and to-do application"
-  homepage "http://joplin.cozic.net"
+  desc "A note taking and to-do application with synchronisation capabilities."
```
Comment (ilovezfs) [inline Formula/joplin.rb]: The description cannot start with `A` or `An` and must not end with a period. :)

**Comment by ilovezfs** [comment]
>I don't mind either way but please confirm which one it should be.

For a substring match, we usually use just a string, which is automatically converted into a regex. So either
```
"editor = subl"
```
or
```
/^editor = subl$/
```
if you want to be more precise

but not
```
/editor = subl/
```

**Comment by laurent22** [comment]
Ok I have updated the assert_match and description.

**Comment by laurent22** [comment]
Hi, just checking if everything is ok with this formula? I made a few more changes the other days but if I missed anything else, feel free to let me know.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @laurent22!

**Comment by laurent22** [comment]
Brilliant thanks @ilovezfs !

**Pull request closed without merging**